### PR TITLE
GRO-599: 2 Factor Authentication migration for settings2/edit-settings

### PR DESCRIPTION
**Pull request by The-Beez-Kneez**
Migrated over the 2FA portion of the `edit-settings` for the new settings2 route

New Behavior:
Screen Shot 2021-11-30 at 9 23 21 AM

### CODE REVIEW

File: src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/__tests__/SettingsEditSettingsTwoFactorAuthTestPage.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
+import { RootTestPage, expectOne } from "v2/DevTools/RootTestPage"
+
+export class SettingsEditSettingsTwoFactorAuthTestPage extends RootTestPage {
```
Comment (dzucconi) [inline src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/__tests__/SettingsEditSettingsTwoFactorAuthTestPage.tsx:3]: This is _weird_ — haven't seen this pattern before

File: src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/__tests__/fixtures.ts [hunk 2]
```diff
@@ -0,0 +1,205 @@
+import { BackupSecondFactorModalContentQueryRawResponse } from "v2/__generated__/BackupSecondFactorModalContentQuery.graphql"
+import { CreateAppSecondFactorMutationResponse } from "v2/__generated__/CreateAppSecondFactorMutation.graphql"
+import { CreateBackupSecondFactorsMutationResponse } from "v2/__generated__/CreateBackupSecondFactorsMutation.graphql"
+import { CreateSmsSecondFactorMutationResponse } from "v2/__generated__/CreateSmsSecondFactorMutation.graphql"
+import { DeliverSecondFactorMutationResponse } from "v2/__generated__/DeliverSecondFactorMutation.graphql"
+import { DisableSecondFactorMutationResponse } from "v2/__generated__/DisableSecondFactorMutation.graphql"
+import { EnableSecondFactorMutationResponse } from "v2/__generated__/EnableSecondFactorMutation.graphql"
+import { TwoFactorAuthenticationQueryRawResponse } from "v2/__generated__/TwoFactorAuthenticationQuery.graphql"
+import { UpdateAppSecondFactorMutationResponse } from "v2/__generated__/UpdateAppSecondFactorMutation.graphql"
+import { UpdateSmsSecondFactorMutationResponse } from "v2/__generated__/UpdateSmsSecondFactorMutation.graphql"
```
Comment (dzucconi) [inline src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/__tests__/fixtures.ts:10]: Where are these files? Presumably they need to be migrated & updated as well.

File: src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/SettingsEditSettingsTwoFactor.tsx [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+      </Text>
+      {secondFactorFragments()}
+      {me.hasSecondFactorEnabled && (
+        <BackupSecondFactorFragmentContainer mt={2} me={me} />
+      )}
+    </Box>
+  )
+}
+
+export const SettingsEditSettingsTwoFactorRefetchContainer = createRefetchContainer(
```
Comment (dzucconi) [inline src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/SettingsEditSettingsTwoFactor.tsx:69]: Wondering how this pattern of using refetch containers started. (If you're using mutations correctly you don't need a refetch)

File: src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/SettingsEditSettingsTwoFactor.tsx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+export interface SettingsEditSettingsTwoFactorProps extends SystemContextProps {
+  me: SettingsEditSettingsTwoFactor_me
+  relay: RelayRefetchProp
+}
+
+export const SettingsEditSettingsTwoFactor: React.FC<SettingsEditSettingsTwoFactorProps> = props => {
+  const { me, relay } = props
+
+  const secondFactorFragments = () => {
+    if (typeof window !== "undefined") {
```
Comment (dzucconi) [inline src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/SettingsEditSettingsTwoFactor.tsx:24]: This shouldn't be necessary and is indicative of some larger problem. Inline this and let's figure out what that is.

File: src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/SettingsEditSettingsTwoFactor.tsx [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+import { SystemContextProps, useSystemContext } from "v2/System"
+import { renderWithLoadProgress } from "v2/System/Relay/renderWithLoadProgress"
+import { SystemQueryRenderer } from "v2/System/Relay/SystemQueryRenderer"
+import { SettingsEditSettingsTwoFactor_me } from "v2/__generated__/SettingsEditSettingsTwoFactor_me.graphql"
+import { SettingsEditSettingsTwoFactorQuery } from "v2/__generated__/SettingsEditSettingsTwoFactorQuery.graphql"
+
+import { SmsSecondFactorFragmentContainer } from "v2/Components/UserSettings/TwoFactorAuthentication/Components/SmsSecondFactor"
+import { BackupSecondFactorFragmentContainer } from "v2/Components/UserSettings/TwoFactorAuthentication/Components/BackupSecondFactor"
+import { AppSecondFactorFragmentContainer } from "v2/Components/UserSettings/TwoFactorAuthentication/Components/AppSecondFactor"
```
Comment (dzucconi) [inline src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/SettingsEditSettingsTwoFactor.tsx:13]: We need to migrate these too. They are improperly styled and appear to have some bugs judging from the guard on  window.

File: src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/SettingsEditSettingsTwoFactor.tsx [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+      environment={relayEnvironment}
+      variables={{}}
+      query={graphql`
+        query SettingsEditSettingsTwoFactorQuery @raw_response_type {
+          me {
+            ...SettingsEditSettingsTwoFactor_me
+          }
+        }
+      `}
+      render={renderWithLoadProgress(
```
Comment (dzucconi) [inline src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/SettingsEditSettingsTwoFactor.tsx:108]: We're going to want to implement a real placeholder.

File: src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/__tests__/SettingsEditSettingsTwoFactorAuthTestPage.tsx (hunk 1 above)
Comment (The-Beez-Kneez) [reply src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/__tests__/SettingsEditSettingsTwoFactorAuthTestPage.tsx:3]: Was going on the naming convention we have been using so far, which has been the `Settings` prefix + the tab section being migrated (in this instance that is `Edit-Settings`.

Comment (dzucconi) [reply src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/__tests__/SettingsEditSettingsTwoFactorAuthTestPage.tsx:3]: Not the naming but rather using some class to wrap a bunch of test related methods. Excessive and I'm not sure what RootTestPage is.

File: src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/__tests__/fixtures.ts (hunk 2 above)
Comment (The-Beez-Kneez) [reply src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/__tests__/fixtures.ts:10]: These are located in that same fixture file. They were only being utilized in this`TwoFactorAuthentication.jest.tx` file, and now replicated for the `SettingsEditSettingsTwoFactor.jest.tsx` files.

File: src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/SettingsEditSettingsTwoFactor.tsx (hunk 3 above)
Comment (The-Beez-Kneez) [reply src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/SettingsEditSettingsTwoFactor.tsx:69]: Not really sure, but they are littered throughout the 2FA and other Settings code from like 2-3 years ago.

File: src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/__tests__/fixtures.ts (hunk 2 above)
Comment (dzucconi) [reply src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/__tests__/fixtures.ts:10]: Sorry, I actually mean the corresponding front end components that these specs are referring to.

File: src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/SettingsEditSettingsTwoFactor.tsx (hunk 4 above)
Comment (The-Beez-Kneez) [reply src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/SettingsEditSettingsTwoFactor.tsx:24]: This was something that @jacobherrington and I paired on to find a workaround. He had seen some similar errors associated with this whilst working within Braze. Tagging him here, cause he had some curiosity as to it.

File: src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/__tests__/SettingsEditSettingsTwoFactorAuthTestPage.tsx (hunk 1 above)
Comment (damassi) [reply src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/__tests__/SettingsEditSettingsTwoFactorAuthTestPage.tsx:3]: this is part of the old testing scaffold that David wrote for order app. It shouldn't be used any more, but in this case all this stuff is just code being relocated.

File: src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/__tests__/fixtures.ts [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+  "xfr93424b1",
+  "a93n5nziu3",
+  "asdf93nz81",
+  "8d300a5493",
+  "q0499zn411",
+  "fn3i1x239m",
+  "asd0893n2d",
+  "a9zmemiejs",
+]
+
```
Comment (damassi) [inline src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/__tests__/fixtures.ts:38]: oh geez, fixtures :(

**Pull request merged**